praisonai/inbuilt_tools/autogen_tools.py

```
@@ -0,0 +1,209 @@
+sys.path.insert(0, os.getcwd())
+root_directory = os.getcwd()
+tools_py_path = os.path.join(root_directory, 'tools.py')
+tools_dir_path = Path(root_directory) / 'tools'
+
+tools_module = None
+
+if os.path.isfile(tools_py_path):
+    logging.info(f"{tools_py_path} exists in the root directory. Loading {tools_py_path} and skipping tools folder.")
+    tools_module = importlib.import_module("tools")
+elif tools_dir_path.is_dir():
+    logging.info(f"tools folder exists in the root directory. Loading {tool_name} from tools/{tool_name}.py.")
```

Ruff (0.15.13)

[error] Undefined name `tool_name`

(F821)